cpu/sam0_common/periph/gpio_ll_irq.c

```diff
@@ -0,0 +1,315 @@
+/*
+ * Copyright (C) 2015 HAW Hamburg
+ *               2016 INRIA
+ *               2022 Otto-von-Guericke-Universität Magdeburg
+ *               2023 Gerson Fernando Budke
+ *               2023 Hugues Larrive
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for more
+ * details.
+ */
+
+/**
+ * @ingroup     cpu_sam0_common
+ * @ingroup     drivers_periph_gpio_ll_irq
+ * @{
+ *
+ * @file
+ * @brief       IRQ implementation of the GPIO Low-Level API for SAM0
+ *
+ * @author      Troels Hoffmeyer <troels.d.hoffmeyer@gmail.com>
+ * @author      Thomas Eichinger <thomas.eichinger@fu-berlin.de>
+ * @author      Kaspar Schleiser <kaspar@schleiser.de>
+ * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
+ * @author      Juergen Fitschen <me@jue.yt>
+ *
+ * @}
+ */
+
+#include <errno.h>
+
+#include "bitarithm.h"
+#include "cpu.h"
+#include "irq.h"
+#include "periph/gpio_ll_irq.h"
+#include "periph_conf.h"
+#include "periph_cpu.h"
+
+#define ENABLE_DEBUG 0
+#include "debug.h"
+
+/**
+ * @brief   Number of external interrupt lines
+ */
+#ifdef CPU_COMMON_SAML1X
+#define IRQS_NUMOF                  (8U)
+#else
+#define IRQS_NUMOF                  (16U)
+#endif
+
+/**
+ * @brief   The GCLK used for clocking EXTI
+ */
+#ifndef CONFIG_SAM0_GCLK_GPIO
+#define CONFIG_SAM0_GCLK_GPIO       (SAM0_GCLK_MAIN)
+#endif
+
+/* Consistify naming */
+#ifndef EIC_SEC
+#define EIC_SEC EIC
+#endif
+
+struct isr_ctx {
+    gpio_ll_cb_t cb;
+    void *arg;
+};
+
+static struct isr_ctx isr_ctx[IRQS_NUMOF];
+
+extern void gpio_ll_mux(gpio_port_t port, uint8_t pin, gpio_mux_t mux);
+
+static int get_exti_num(unsigned port_num, uint8_t pin)
+{
+    if (port_num >= ARRAY_SIZE(exti_config)) {
+        return -1;
+    }
```

Comment:
Can this be an `assert()`?

Reply:
The call sites of `get_exti_num()` actually do:

```C
    assume((unsigned)exti_num < IRQS_NUMOF);
```

That should cover this as well, but also gives the compiler information about what values are to be expected within `exti_config`.